projects/optic/src/commands/oas/shapes/patches/generators/type.ts

```diff
@@ -76,32 +78,108 @@ export function* typePatches(
         // handles removal of keys that are no longer allowed
         value: Schema.merge(
           currentPropertySchema,
-          Schema.baseFromValue(diff.example)
+          Schema.baseFromValue(diff.example, openAPIVersion)
         ),
       }),
     ];
   }
 
-  // option one: convert to a one-off
-  yield {
-    description: `make ${diff.key} oneOf`,
-    diff,
-    impact: [
-      PatchImpact.Addition,
-      !shapeContext.location
-        ? PatchImpact.BackwardsCompatibilityUnknown
-        : 'inRequest' in shapeContext.location
-        ? PatchImpact.BackwardsCompatible
-        : PatchImpact.BackwardsIncompatible,
-    ],
-    groupedOperations: makeOneOfOperations(),
-  };
+  if (openAPIVersion === '3.0.x') {
```

Comment:
Can we write tests for these 4 different code paths?

Reply:
So there are two existing tests which have been updated for when a field becomes nullable. And another two when a null field is first seen